consensus/src/runway/mod.rs

```diff
@@ -885,17 +886,18 @@ pub(crate) async fn run<H, D, US, UL, MK, DP, FH, SH>(
     let alerter_keychain = keychain.clone();
     let alert_messages_for_network = network_io.alert_messages_for_network;
     let alert_messages_from_network = network_io.alert_messages_from_network;
+
+    let mut alerter_service = Service::new(
```

Comment:
I'm pretty sure you actually want to be able to separately construct the `Handler` and use it for any "simulations", rather than the `Service`. You don't want to be pushing into IO channels during this preprocessing. This would also let you have `Service` with only `run` (and maybe `new`) public.